packages/xo-web/src/common/xo/index.js

```diff
@@ -1328,13 +1328,17 @@ export const snapshotVms = vms =>
     icon: 'memory',
     title: _('snapshotVmsModalTitle', { vms: vms.length }),
     body: <SnapshotVmModalBody vms={vms} />,
-  }).then(
-    ({ names, saveMemory, descriptions }) =>
-      Promise.all(
-        map(vms, vm => snapshotVm(vm, names[vm], saveMemory, descriptions[vm]))
-      ),
-    noop
-  )
+  })
+    .then(
+      ({ names, saveMemory, descriptions }) =>
+        Promise.all(
+          map(vms, vm =>
+            snapshotVm(vm, names[vm], saveMemory, descriptions[vm])
+          )
+        ),
+      noop
+    )
+    .catch(e => error(_('snapshotError'), e.message))
```

Comment:
Better throw the error in case if we want to handle it in the high chain call level. You can use [`tapCach`](https://github.com/JsCommunity/promise-toolbox#promisetapcatchonrejected).

Reply:
That's actually something we discussed (where to handle errors, how to avoid multiple error notifications) and we came to the conclusion that we need to decide for each method if the error should be handled here or later, based on how and where it's called, since there isn't a trivial generic answer. So I would actually be fine having a hard `catch` here and nothing later, if that's what we want for this one.